### DISCUSSION TITLE
Branch not displayed in terminal prompt

### DISCUSSION THREAD

**Discussion by maiertech** [community/community#3563] [Codespaces]
I noticed that when I branch off a commit that also has a tag, the terminal prompt in my codespace displays the tag instead of the branch. Here is an example:

<img width="934" alt="Screen Shot 2021-03-16 at 9 48 04 PM" src="https://user-images.githubusercontent.com/1482402/111402442-973ac580-86a1-11eb-8c6d-063ffa7b04b7.png">

In this example I am on branch `maiertech/issue256` but the terminal prompt displays the tag. This is what the commit I am at looks like:

<img width="820" alt="Screen Shot 2021-03-16 at 9 49 11 PM" src="https://user-images.githubusercontent.com/1482402/111402576-d0733580-86a1-11eb-95dd-7bc426f8e3e9.png">

**Comment by PranamBhat** [comment]
Hi @maiertech 

Make sure your branch has a valid name and check the settings if you have made any changes.

Refresh your repository/github branch. Restart your terminal.

**Comment by jkeech** [comment]
The bash prompt can be seen at https://github.com/microsoft/vscode-dev-containers/blob/master/script-library/common-debian.sh#L257. I suspect the command `git describe --contains --all HEAD 2>/dev/null` is returning the tag in your case. Can you confirm? Should be easy to update the command to something better to get the actual branch that's checked out 😄

**Reply by jkeech** [reply]
Filed https://github.com/microsoft/vscode-dev-containers/issues/793

**Reply by maiertech** [reply]
Yes, `git describe --contains --all HEAD 2>/dev/null` return the tag.

When will above fix make it's way into container images?

**Reply by jkeech** [reply]
@maiertech, the PR is merged already. It looks like this will be included in the upcoming 1.3.0 release of the codespaces image, which should roll out across the service within a week.
